### DISCUSSION TITLE
Allow using a non-default branch as the production environment

### DISCUSSION THREAD

**Discussion by shc023** [vercel/vercel#4862] [Ideas] [answered]
While setting up a project in the Vercel dashboard, I'm unable to find a way to work the Github Integration into our existing git flow.

Our default branch is usually `master` which we use as staging, and developers PR against `master` to first land the code on staging and get QA'ed. We have a separate `production` branch that we push into once in a while to do production releases.

However, since Vercel uses the default branch as production, we're now forced to change our git flow and have to make "production" the default branch, and everyone has to be extremely careful while pushing, having to remember to always manually select "master" to PR against. If we don't switch default branch, there seems to be no way to use the "Production Environment Variables" for a non-default branch.

Is there an existing solution in the Vercel Dashboard that I'm missing? If not, it seems like it would make sense to allow users to select which branch they want to treat as "production", instead of locking in on the "default branch" in the github config.

**Comment by mcsdevv** [comment, marked as answer]
Hey @shc023, I'm pleased to say this can now be done!

Further information: https://vercel.com/blog/custom-production-branch
